src/include/s390_gcm.h

```diff
@@ -681,8 +681,8 @@ static inline int s390_aes_gcm_simulate_kma_intermediate(const unsigned char *in
 	if (rc)
 		return rc;
 
-	ctx->total_aad_length += aad_length;
-	ctx->total_input_length += data_length;
+	ctx->total_aad_length += aad_length * 8;
```

Comment:
So it seems that CPACF actually would support AAD and input text on bit boundary? 
However, the libica API only supports passing AAD and input text on byte boundary? 

I guess this is OK (and we would need an API break to add support for GCM on bit boundary anyway....).

Reply:
I'm not sure: NIST says: 

> Although GCM is defined on bit strings, the bit lengths of the plaintext, the AAD, and the IV
> shall all be multiples of 8, so that these values are byte strings.

https://nvlpubs.nist.gov/nistpubs/legacy/sp/nistspecialpublication800-38d.pdf
(page 8)

Reply:
OK, strange that CPACF then defines these fields as number of bits....

Also, doesn't this change mean that it should have calculated a wrong GCM output before? 
Don't we have known-answer-tests for GCM? They should have catched that, right? If we don't have them, maybe its time to introduce them now.

Reply:
Yes, we have gcm known answer tests, but all with byte lengths ... not sure if there are NIST test vectors with bit lengths, as NIST itself says that the lengths should be byte lengths. In libica we enforce byte lengths anyway, but must multiply with 8 for the CPACG parm block.

Reply:
Yes, good point! In fact, when calculating the final tag in s390_gcm_authenticate_last, the lengths are multiplied by 8, so it's correct to leave them as byte values in the ctx.

Reply:
I removed the related commit.